pkg/principalresolver/principalresolver.go

```diff
@@ -11,7 +12,10 @@ type resolver struct {
 	mapping map[did.DID]did.DID
 }
 
-func (r *resolver) ResolveDIDKey(input did.DID) (did.DID, validator.UnresolvedDID) {
+var _ validator.PrincipalResolver = (*resolver)(nil)
+
+func (r *resolver) ResolveDIDKey(ctx context.Context, input did.DID) (did.DID, validator.UnresolvedDID) {
+	// ctx is unused; this implementation only looks in a local mapping.
```

Comment:
No change required here, but a pattern I've used/liked for methods with unused arguments is:
```suggestion
func (r *resolver) ResolveDIDKey(_ context.Context, input did.DID) (did.DID, validator.UnresolvedDID) {
```